stack/dispatcher.go

```diff
@@ -69,8 +69,14 @@ func (d *Dispatcher) CheckTx(ctx basecoin.Context, store state.KVStore, tx basec
 	if err != nil {
 		return res, err
 	}
-	// TODO: check on callback
-	cb := d
+
+	// make sure no monkey business with the context
+	cb := secureCheck(d, ctx)
+
+	// and isolate the permissions and the data store for this app
+	ctx = withApp(ctx, r.Name())
+	store = stateSpace(store, r.Name())
```

Comment:
awesome this was the line I was looking for - would it not be possible for an a basecoin "plugin" to implement their own dispatcher that ignored this Aka allowed for monkey biz? How to we enforce that this dispatcher is used?